server/src/main/java/org/apache/iotdb/db/mpp/execution/executor/RegionWriteExecutor.java

```diff
@@ -348,14 +349,17 @@ public RegionExecutionResult visitCreateMultiTimeSeries(
             }
 
             // filter failed measurement and keep the rest for execution
-            for (Map.Entry<Integer, MetadataException> failingMeasurement :
-                failingMeasurementMap.entrySet()) {
-              entry.getValue().removeMeasurement(failingMeasurement.getKey());
-              LOGGER.error("Metadata error: ", failingMeasurement.getValue());
+            List<Integer> failingMeasurementIndexList =
+                failingMeasurementMap.keySet().stream().sorted().collect(Collectors.toList());
+            int removedNum = 0;
+            for (Integer index : failingMeasurementIndexList) {
+              entry.getValue().removeMeasurement(index - removedNum);
```

Comment:
This operation has a strong dependency on the index values in `failingMeasurementIndexList` which should be sorted...